src/main/java/io/smartcat/ranger/core/DiscreteWeightedValue.java

```diff
@@ -0,0 +1,83 @@
+package io.smartcat.ranger.core;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.math3.distribution.EnumeratedDistribution;
+import org.apache.commons.math3.util.Pair;
+
+/**
+ * Randomly selects one of the provided values following the provided weights.
+ *
+ * @param <T> Type this value would evaluate to.
+ */
+public class DiscreteWeightedValue<T> extends Value<T> {
```

Comment:
Yes we can. Just wondering, what would be the benefit? We would use it on array/list to return desired value. It looks the same as current solution.